src/convert_image/x86.rs

```diff
@@ -428,6 +428,127 @@ fn lrgb_to_i420(
     true
 }
 
+#[inline(always)]
+fn lrgb_to_i444(
+    width: u32,
+    height: u32,
+    _last_src_plane: usize,
+    src_strides: &[usize],
+    src_buffers: &[&[u8]],
+    last_dst_plane: usize,
+    dst_strides: &[usize],
+    dst_buffers: &mut [&mut [u8]],
+    channels: PixelFormatChannels,
+    colorimetry: Colorimetry,
+    sampler: Sampler,
+) -> bool {
+    if last_dst_plane != 2
+        || (last_dst_plane >= dst_strides.len())
+        || (last_dst_plane >= dst_buffers.len())
+        || src_strides.is_empty()
+        || src_buffers.is_empty()
+    {
+        return false;
+    }
+
+    let depth = channels as usize;
+    let col_count = width as usize;
+    let line_count = height as usize;
+    let packed_rgb_stride = depth * col_count;
+
+    let rgb_stride = if src_strides[0] == 0 {
+        packed_rgb_stride
+    } else {
+        src_strides[0]
+    };
+
+    let y_stride = if dst_strides[0] == 0 {
+        col_count
+    } else {
+        dst_strides[0]
+    };
+
+    let u_stride = if dst_strides[1] == 0 {
+        col_count
+    } else {
+        dst_strides[1]
+    };
+
+    let v_stride = if dst_strides[2] == 0 {
+        col_count
+    } else {
+        dst_strides[2]
+    };
+
+    let rgb_plane = &src_buffers[0];
+    let (y_plane, uv_plane) = dst_buffers.split_at_mut(1);
+    let (u_plane, v_plane) = uv_plane.split_at_mut(1);
+
+    let y_plane = &mut y_plane[0][..];
+    let u_plane = &mut u_plane[0][..];
+    let v_plane = &mut v_plane[0][..];
```

Comment:
Have you tried the following?
```rust
    let y_plane = &mut dst_buffers[0][..];
    let u_plane = &mut dst_buffers[1][..];
    let v_plane = &mut dst_buffers[2][..];
```

Reply:
It does not compile.
error[E0499]: cannot borrow `*dst_buffers[_]` as mutable more than once at a time
   --> src\convert_image\x86.rs:485:24
    |
484 |     let y_plane = &mut dst_buffers[0][..];
    |                        -------------- first mutable borrow occurs here
485 |     let u_plane = &mut dst_buffers[1][..];
    |                        ^^^^^^^^^^^^^^ second mutable borrow occurs here
...
501 |     if y_stride * line_count > y_plane.len()
    |                                ------- first borrow later used here

error[E0499]: cannot borrow `*dst_buffers[_]` as mutable more than once at a time
   --> src\convert_image\x86.rs:486:24
    |
484 |     let y_plane = &mut dst_buffers[0][..];
    |                        -------------- first mutable borrow occurs here
485 |     let u_plane = &mut dst_buffers[1][..];
486 |     let v_plane = &mut dst_buffers[2][..];
    |                        ^^^^^^^^^^^^^^ second mutable borrow occurs here
...
501 |     if y_stride * line_count > y_plane.len()
    |                                ------- first borrow later used here